lib/ansible/modules/remote_management/oneview/oneview_fc_network_facts.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/python
+# Copyright (c) 2016-2017 Hewlett Packard Enterprise Development LP
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: oneview_fc_network_facts
+short_description: Retrieve the facts about one or more of the OneView Fibre Channel Networks
+description:
+    - Retrieve the facts about one or more of the Fibre Channel Networks from OneView.
+version_added: "2.4"
+requirements:
+    - "hpOneView >= 2.0.1"
+author:
+    - Felipe Bulsoni (@fgbulsoni)
+    - Thiago Miotto (@tmiotto)
+    - Adriane Cardozo (@adriane-cardozo)
+options:
+    name:
+      description:
+        - Fibre Channel Network name.
+      required: false
+
+extends_documentation_fragment:
+    - oneview
+    - oneview.factsparams
+'''
+
+EXAMPLES = '''
+- name: Gather facts about all Fibre Channel Networks
+  oneview_fc_network_facts:
+    config: '/etc/oneview/oneview_config.json'
+  delegate_to: localhost
```

Comment:
The HPE Appliance is REST API based and should be accessible from the node where ansible is running.
The module will then get the URL and user credentials from the config file and use those to login on the appliance and manage it.
It will usually be run on localhost, but can be run on other nodes through ssh, as long as they have connectivity to the HPE OneView appliance.